inc/Addon/GoogleTracking/GoogleTagManager.php

```diff
@@ -68,18 +64,21 @@ class Google_Tag_Manager extends Abstract_Busting {
 	 *
 	 * @since  3.1
 	 * @access public
-	 * @author Remy Perona
 	 *
-	 * @param string           $busting_path Path to the busting directory.
-	 * @param string           $busting_url  URL of the busting directory.
-	 * @param Google_Analytics $ga_busting   A Google_Analytics instance.
+	 * @param string               $busting_path Path to the busting directory.
+	 * @param string               $busting_url  URL of the busting directory.
+	 * @param GoogleAnalytics      $ga_busting   A GoogleAnalytics instance.
+	 * @param WP_Filesystem_Direct $filesystem   Instance of the filesystem handler.
 	 */
-	public function __construct( $busting_path, $busting_url, Google_Analytics $ga_busting ) {
+	public function __construct( $busting_path, $busting_url, GoogleAnalytics $ga_busting, $filesystem = null ) {
 		$blog_id            = get_current_blog_id();
 		$this->busting_path = $busting_path . $blog_id . '/';
 		$this->busting_url  = $busting_url . $blog_id . '/';
 		$this->ga_busting   = $ga_busting;
-		$this->filesystem   = \rocket_direct_filesystem();
+		if ( is_null( $filesystem ) ) {
+			$filesystem = rocket_direct_filesystem();
+		}
+		$this->filesystem = $filesystem;
```

Comment:
```suggestion
		if ( is_null( $filesystem ) ) {
			$this->filesystem = rocket_direct_filesystem();
		} else {
		    $this->filesystem = $filesystem;
		}
```

Reply:
I think this from performance wise is the same, why do we need to adjust that?

Reply:
How about:

`$this->filesystem = is_null( $filesystem ) ? rocket_direct_filesystem() : $filesystem;`

Makes the code more readable

Reply:
Ok, Nice from readability perspective